common/types/epoch.go

```diff
@@ -11,49 +11,53 @@ import (
 // EpochID is the running epoch number. It's zero-based, so the genesis epoch has EpochID == 0.
 type EpochID uint32
 
+func (e EpochID) Uint32() uint32 {
+	return uint32(e)
+}
+
 // EncodeScale implements scale codec interface.
-func (l EpochID) EncodeScale(e *scale.Encoder) (int, error) {
-	n, err := scale.EncodeCompact32(e, uint32(l))
+func (e EpochID) EncodeScale(enc *scale.Encoder) (int, error) {
+	n, err := scale.EncodeCompact32(enc, uint32(e))
 	if err != nil {
 		return 0, err
 	}
```

Comment:
I know this is unrelated to your changes, but I think this is not correct. `n` might be 0 but it also might not be (if some bytes were written to `enc`).